sdk/python/feast/feature_store.py

```diff
@@ -477,11 +477,15 @@ def _get_feature_view(
         name: str,
         hide_dummy_entity: bool = True,
         allow_registry_cache: bool = False,
-    ) -> FeatureView:
+    ) -> BaseFeatureView:
         feature_view = self._registry.get_feature_view(
             name, self.project, allow_cache=allow_registry_cache
         )
-        if hide_dummy_entity and feature_view.entities[0] == DUMMY_ENTITY_NAME:
+        if (
+            isinstance(feature_view, FeatureView)
```

Comment:
`utils._list_feature_views` has a similar logic but misses the `isinstance` check, don't we need it there too?

Reply:
Not yet, it's using `list_feature_views` to get the objects which is still `FeatureView` specific, this PR only makes changes to `get_feature_view`. I'll do that in a follow-up.